atlp-rwanda/atlp-devpulse-fn
# #247  Applicant upload documents in Cycles Application

# Pull Request Description

## Feature: #247  Applicant upload documents in Cycles Application

### What does this PR do?


This PR adds the feature allowing the trainees to upload documents when applying to any application cycles. Documents uploaded are ID Card, Resume and Cover letter. These documents are uploaded so that the admin can use them when reviewing applications and the user have the ability to view those documents.


### How Can This Be Manually Tested?
1. Login as a trainee
- Click on my application
- Apply on a cycle by providing all required information and all documents
- After the application is succed you have to go back on my applications
- Click on track my application
- You can see all the informations you submitted and then you can click on any document to view it.
- The new tab with the document will opened directly.

2. Login as an Admin
- Login as Admin and click on applications cycles
- Click on any specific cycle you want to review in
- For every applicant see in the documents tab so that you can view the document they uploaded. Click on any document and view it's details in new tab.

### Credentials:
**Admin:**
email: admin@example.com
password: password123

**Trainee:**
email: jane@example.com
password: password123



# Issue Nnumber:
#247 



### ScreenShots:

![Screenshot 2024-11-20 221750](https://github.com/user-attachments/assets/fba0a45e-3d38-4f06-a745-58b296d00042)
![Screenshot 2024-11-21 143546](https://github.com/user-attachments/assets/22f22d15-23b3-4c99-9346-e59fe222d920)
![Screenshot 2024-11-20 221805](https://github.com/user-attachments/assets/89b45b79-ce72-4ffe-8773-358b6e6c1dc1)
![Screenshot 2024-11-20 221822](https://github.com/user-attachments/assets/fae01a82-a3d3-44ef-80ea-cd672af753ac)
